test wdt_basic_api failed on nucleo_f746zg

FRASTM opened the pull request:
This patches add a delay after setting the watchdog to wait for the register (Prescaler and Counter registers) to be updated and reload before leaving.
It modifies  the `iwdg_stm32_install_timeout` function of the `wdt_iwdg_stm32.c` 

Fixes  #22257

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on drivers/watchdog/wdt_iwdg_stm32.c:
```diff
@@ -126,6 +126,17 @@ static int iwdg_stm32_install_timeout(struct device *dev,
 
+	/* Wait for the register to be updated */
+	while (LL_IWDG_IsReady(iwdg) != 0) {
+		/* Wait until WVU, RVU, PVU are reset before updating  */
```
I think it should be "Wait until WVU, RVU, PVU are **set** before updating"

FRASTM replied:
Indeed, the loop waits for the LL_IWDG_IsReady function to return True, meaning that update is completed (The bits are set "to indicate that an update is ongoing")

erwango reviewed (changes requested):
One minor comment.
Btw, is there a potential clean up to do in testcase.yaml file to extend testing  on more targets ? (or clean up unrequired tweaks on nucleo_l496 board)

ioannisg commented:
@FRASTM will you fix this for the upcoming 2.2. release?

FRASTM committed:
driver: watchdog: stm32 install watchdog

This patches add a delay after setting the watchdog
to wait for the register (Prescaler and Counter registers)
to be updated before leaving (until LL_IWDG_IsReady is true)

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
There  is no need to wait for LL_IWDG_IsReady(iwdg)
before setting the Prescaler and the reload counter. The sequence to wait after writing to the registers that the update is complete.

FRASTM commented:
There  is no need to wait for `LL_IWDG_IsReady(iwdg)`
before setting the Prescaler and the reload counter. The sequence to wait after writing to the registers that the update is complete.
